torch/_dynamo/symbolic_convert.py

```diff
@@ -611,29 +611,30 @@ def step(self):
         )
 
     def run(self):
-        try:
-            self.output.push_tx(self)
-            while (
-                self.instruction_pointer is not None
-                and not self.output.should_exit
-                and self.step()
-            ):
-                pass
-        except BackendCompilerFailed:
-            raise
-        except Exception as e:
-            if config.replay_record_enabled:
-                e.exec_record = self.exec_recorder.get_record()  # type: ignore[attr-defined]
-            raise
-        finally:
-            self.output.pop_tx()
-            # Cleanup the outputGraph to delete the held tensors. We perform the
-            # cleanup only for InstructionTranslator and not
-            # InliningInstructionTranslator. The InliningInstructionTranslator
-            # mutates the output object and is restored to original state if
-            # there was an exception.
-            if isinstance(self, InstructionTranslator):
-                self.output.cleanup()
+        with TracingContext.current_frame(self.frame_summary()):
+            try:
+                self.output.push_tx(self)
+                while (
+                    self.instruction_pointer is not None
+                    and not self.output.should_exit
+                    and self.step()
```

Comment:
Cool! Just to confirm, then, `TracingContext.get().frame_summary_stack` should give me the "current" stack of user functions that we're tracing, right?

So this won't give the current instruction line number...is there a way to get that? Should we have an inner context manager per `step()` call?

Reply:
It should give the line no for the frame, which is what you wanted, I think, to start.

The problem with instruction line number is that isn't really a notion? Like, there isn't a great mapping of one to the other as far as I am aware.

Reply:
We can, however, take a stab at building something that gives you mapping of lines within the frame, perhaps? I'd need to look more...

Reply:
Perhaps we can start with the logic that guards the following in `step()`, which I believe marks out the beginning of an instruction sequence mapping to a user code line:
```
if inst.starts_line and self.lineno != inst.starts_line:
            self.lineno = inst.starts_line
            log.debug(f"TRACE starts_line {self.f_code.co_filename}:{self.lineno}")
```

Reply:
Sure! Whatever you like to help you get more data.

That being said - I think there might be a lot of value in collecting this data in a structure, and THEN dumping it, kinda like in the structure we set up in this PR. Chances are, it will be more valuable to see a whole set of frames / line numbers that led to a given point for provenance / debugging than to just get dumps of linenos.